controllers/component_controller.go

```diff
@@ -456,6 +500,36 @@ func getInitialBuildWorkspaceSubpath() string {
 	return "initialbuild-" + time.Now().Format("2006-Feb-02_15-04-05")
 }
 
+// getGitProvider takes a Git URL of the format https://github.com/foo/bar and returns
+// https://github.com
+func getGitProvider(gitURL string) (string, error) {
+	u, err := url.Parse(gitURL)
+
+	// We really need the format of the string to be correct.
+	// We'll not do any autocorrection.
+	if err != nil || u.Scheme == "" {
+		return "", fmt.Errorf("failed to parse string into a URL: %v or scheme is empty", err)
+	}
+	return u.Scheme + "://" + u.Host, nil
+}
+
+func updateServiceAccountIfSecretNotLinked(ctx context.Context, sourceSecretName string, serviceAccount *corev1.ServiceAccount) bool {
+	isSecretPresent := false
+	for _, credentialSecret := range serviceAccount.Secrets {
+		if credentialSecret.Name == sourceSecretName {
```

Comment:
Nit: Could you add a case or two for this functions test that covers lines 519-525 (what codecov is complaining about)?